default.nix

```diff
@@ -36,139 +40,81 @@ let iosSupport =
         });
       };
     };
+    androidPICPatches = self: super: {
+      haskell = super.haskell // {
+        compiler = super.haskell.compiler // {
+          integer-simple = super.haskell.compiler.integer-simple // {
+            ghc843 = super.haskell.compiler.integer-simple.ghc843
+                     .overrideAttrs (drv: {
+              patches = (drv.patches or [])
+                      ++ [ ./android/patches/force-relocation.patch ];
+            });
+          };
+        };
+      };
+    };
     nixpkgs = nixpkgsFunc ({
       inherit system;
       overlays = [globalOverlay];
       config = {
-        allowUnfree = true;
-        allowBroken = true; # GHCJS is marked broken in 011c149ed5e5a336c3039f0b9d4303020cff1d86
         permittedInsecurePackages = [
           "webkitgtk-2.4.11"
         ];
         packageOverrides = pkgs: {
-          webkitgtk = pkgs.webkitgtk216x;
+          webkitgtk = pkgs.webkitgtk220x;
           # cabal2nix's tests crash on 32-bit linux; see https://github.com/NixOS/cabal2nix/issues/272
           ${if system == "i686-linux" then "cabal2nix" else null} = pkgs.haskell.lib.dontCheck pkgs.cabal2nix;
         };
+
+        # XCode needed for native macOS app
+        # Obelisk needs it to for some reason
+        allowUnfree = true;
```

Comment:
We can use `allowUnfreePredicate` to make sure we only allow exactly what we want.

Reply:
You can definitely still override it in the ```config``` argument using something like:

```
nix-build --arg config "{ alllowUnfree = false; allowUnfreePredicate = ...; }"
```

There's enough unfree stuff in reflex-platform that it will be a little difficult to list it all.

Reply:
This seems reasonable to turn into another ticket.